### DISCUSSION TITLE
Expose Buttons to Remote Send Command Services

### DISCUSSION THREAD

**Discussion by brandon-meadows** [PRProd/HA-Firemote#102] [Q&A] [answered]
It would be a nice feature to use some of the button presses ( left, right, up, down, etc.) as remote commands that could be used in automations and scripts. Is it possible to expose them?

**Comment by PRProd** [comment, marked as answer]
@brandon-meadows - Hello!  Thanks for writing!

If I'm understanding your question correctly, you're asking for the ability to use Firemote button presses in other HA automatons, is that correct?  If so, a brief explanation about how the Firemote Card works might help.

HA-Firemote is a front end Home Assistant Card that relies upon already existing Home Assistant integrations to make things happen, such as the Android Debug Bridge (formerly known as Android TV) and soon, the Android TV Remote Integration.  The commands for up, down, left, right, launch this app, select this thing, etc. are all sent as commands from the Firemote card to a Home Assistant integration or service in almost the exact same way that you would send those commands in a script or automation.  If you are looking to add similar commands to an automation, it's not the Firemote card that you want to hook into, but instead it's the same integrations and services that the Firemote card uses that you want to emulate and use for yourself.

Am I understanding your question correctly?  If so, please allow me to help you create any type of automation you want, and show you how to use the work that I've done with the Firemote card to copy/paste your way to anything else you'd like to do.

**Reply by ovsims** [reply]
I'm looking to do the exact thing but I can not figure it out. Specifically. I use Alexa to run my scripts to turn on my TV, nvidia shield, close the blind and dim the lights. I cannot figure out how to get my scripts to auto launch an app. Or say press the button on the Firemote card. currently I have these on my remote and I would like to say Alexa watch _____ and have my script launch one of them. I plan on setting up a wall tablet as well and would like to press the button on your Firemote and have it run the script to launch the app and dim the lights, close the blind etc. if that's possible. will you help me figure it out?

![image](https://github.com/PRProd/HA-Firemote/assets/113260214/07ee5301-ad67-4609-8089-ef90787678f8)

**Comment by PRProd** [comment]
@brandon-meadows - Can I offer any additional help for you here?

**Reply by brandon-meadows** [reply]
Thanks. I was able to figure it out with your explanation above.
